cli/src/main/java/com/devonfw/tools/ide/io/FileAccessImpl.java

```diff
@@ -282,22 +284,74 @@ private void copyRecursive(Path source, Path target, FileCopyMode mode) throws I
     }
   }
 
+  @Override
+  public void makeSymlinkRelative(Path link) {
+
+    makeSymlinkRelative(link, false);
+  }
+
+  @Override
+  public void makeSymlinkRelative(Path link, boolean followTarget) {
+
+    if (!Files.isSymbolicLink(link)) {
+      throw new IllegalStateException(
+          "Can't call makeSymlinkRelative on " + link + " since it is not a symbolic link.");
+    }
+    Path linkTarget = null;
+    try {
+      linkTarget = followTarget ? link.toRealPath() : Files.readSymbolicLink(link);
+    } catch (IOException e) {
+      throw new RuntimeException("For link " + link + " the call to "
+          + (followTarget ? "toRealPath" : "readSymbolicLink") + " in method makeSymlinkRelative failed.", e);
+    }
+    this.context.getFileAccess().delete(link); // delete old absolute link
+    this.context.getFileAccess().relativeSymlink(link, linkTarget); // and replace it by the new relative link
```

Comment:
it is pointless to call yourself via `this.context.getFileAccess().` - simply remove this prefix before `delete` or `relativeSymlink`.